### PR TITLE
docs: Update README with class/field docs and some other improvements…

### PR DESCRIPTION
**Pull request by scottmcmaster**
…\rTESTING=manual

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,11 +64,250 @@ export const promotedClient = newPromotedClient({
...
+| ------------------------------ | -------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `deliveryClient`               | ApiClient                                                      | API client to make a POST request to the Delivery API endpoint including the `x-api-key` header, endpoint and header value obtained from Promoted                                                                                                                                         |
+| `metricsClient`                | ApiClient                                                      | API client to make a POST request to the Delivery API endpoint including the `x-api-key` header, endpoint and header value obtained from Promoted                                                                                                                                         |
```
Comment (danbosnichill) [inline README.md:72]: s/Delivery/Metrics/

File: README.md [hunk 2]
```diff
@@ -58,11 +64,250 @@ export const promotedClient = newPromotedClient({
...
+}
+```
 
+### UserInfo
+
+Basic information about the request user.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+`userId` | String | Yes | The platform user id, cleared from Promoted logs.
+`logUserId` | String | Yes | A different user id (presumably a UUID) disconnected from the platform user id, good for working with unauthenticated users or implementing right-to-be-forgotten.
```
Comment (danbosnichill) [inline README.md:99]: Some of the platforms call this an anonymousUserId.

File: README.md [hunk 3]
```diff
@@ -58,11 +64,250 @@ export const promotedClient = newPromotedClient({
...
+Content being served at a certain position.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+`userInfo` | UserInfo | Yes | The user info structure.
+`insertionId` | String | Yes | Generated by the SDK (_do not set_)
+`requestId` | String | Yes | Generated by the SDK when needed (_do not set_)
+`contentId` | String | No | Identifier for the content to be ranked, must be set.
+`retrievalRank` | Number | Yes | Optional original ranking of this content item.
+`retrievalScore` | Number | Yes | Optional original quality score of this content item.
+`properties` | Properties | Yes | Any additional custom properties to associate. For v1 integrations, it is fine not to fill in all the properties.
```
Comment (danbosnichill) [inline README.md:142]: We should get a common doc for where to send certain field values.  I'll work on a doc later today and we can link it here.

**Review by danbosnichill — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (scottmcmaster) [reply README.md:99]: Noted there

**Pull request merged**